docs/content/docs/dev/datastream/fault-tolerance/checkpointing.md

```diff
@@ -277,11 +277,10 @@ task with the number of new subtasks equal to the number of running tasks.
 
 ### Waiting for the final checkpoint before task exit
 
-To ensure all the records could be committed for operators using the two-phase commit, 
-the tasks would wait for the final checkpoint completed successfully after all the operators finished. 
-It needs to be noted that this behavior would prolong the execution time of tasks. 
-If the checkpoint interval is long, the execution time would also be prolonged largely. 
-For the worst case, if the checkpoint interval is set to `Long.MAX_VALUE`, 
-the tasks would in fact be blocked forever since the final checkpoint would never happen.
+To ensure all the records could be committed for operators using the two-phase commit,
+the tasks would wait for the final checkpoint completed successfully after all the operators finished.
+The final checkpoint would be triggered immediately after all operators have reached end of data, 
+without waiting for periodic triggering.
```

Comment:
nit:

> without waiting for periodic triggering.

->

> without waiting for periodic triggering, but the job will need to wait for this final checkpoint to be completed.

?

Reply:
Thanks for the review.
Do we really need to add this explanation? As it has been mentioned before.
```
the tasks would wait for the final checkpoint completed successfully after all the operators finished.
```

Reply:
Probably not in that case.